src/RefractDataStructure.cc

```diff
@@ -873,30 +875,121 @@ namespace drafter
             }
         };
 
+        template <>
+        struct AllElementsToAtribute<refract::EnumElement> {
+            using T = refract::EnumElement;
+
+            template <typename U>
+            void operator()(const U& values, const std::string& key, refract::IElement* element)
+            {
+
+                auto merged = Merge<T>()(values);
+                auto const& items = std::get<0>(merged);
+
+                if (items.empty()) {
+                    return;
+                }
+
+                refract::ArrayElement* a = new refract::ArrayElement;
+                for (auto const& item : items) {
+                    refract::EnumElement* e = new refract::EnumElement;
+                    e->set(item);
+                    a->push_back(e);
+                }
+
+                element->attributes[key] = a;
+            }
+        };
+
         template <typename T>
         struct LastElementToAttribute {
 
             template <typename U>
-            void operator()(const U& values, const std::string& key, refract::IElement* element)
+            void operator()(
+                const U& values, const std::string& key, refract::IElement* element, ConversionContext& /* context */)
             {
 
                 if (values.empty()) {
                     return;
                 }
 
+                auto rbegin = values.rbegin();
+
+                // pick last one
                 ElementInfoToElement<T> fetch;
-                refract::IElement* value = fetch(*values.rbegin());
+                refract::IElement* value = fetch(*rbegin);
 
-                if (values.size() > 1) {
-                    auto begin = values.rbegin();
-                    begin++; // avoid last one
-                    ReleaseStoredData<T>()(begin, values.rend());
-                }
+                // and release rest of them
+                rbegin++;
+                ReleaseStoredData<T>()(rbegin, values.rend());
 
                 element->attributes[key] = value;
             }
         };
 
+        template <typename T>
+        void CheckForMultipleDefaultDefinitions(const T& values, ConversionContext& context)
+        {
+            if (values.empty()) {
+                return;
+            }
+
+            auto const& first = std::get<0>(*values.begin());
+            // we are sure `first` is valid because precondition check `values.empty()`
+            if ((values.size() > 1) || (first.size() > 1)) {
+
+                mdp::CharactersRangeSet location;
+                for (auto const& item : values) {
```

Comment:
Use exclusively either `const T&` or `T const&` to increase readability. I suggest `const T&` because the const keyword is right in front of what it is bound to.